packages/kbn-esql-ast/src/antlr/esql_lexer.g4

```diff
@@ -4,11 +4,58 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+lexer grammar esql_lexer;
 
-// DO NOT MODIFY THIS FILE BY HAND. IT IS MANAGED BY A CI JOB.
+@header {
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+}
 
-lexer grammar esql_lexer;
-  options { caseInsensitive = true; }
+options {
+  superClass=lexer_config;
```

Comment:
What does it do?

Reply:
It gives the generated lexer a super class. You can see the inheritance [here](https://github.com/elastic/kibana/pull/192027/files#diff-83ea2b95314b572a02f8c32dd305af5841b3f05a79da17e53ed569941af18dcbR25). The same thing happens for the parser.

The `this.isDevVersion()` calls in the grammar (e.g. `DEV_INLINESTATS : {this.isDevVersion()}? 'inlinestats' -> pushMode(EXPRESSION_MODE);`) get embedded in the generated parser and lexer as `if` statements (e.g. [here](https://github.com/elastic/kibana/pull/192027/files#diff-7297fe9e531b4f6aa54553eadeaa93783dcdf4b9f593f1a00b84ce3f6b13b8cdR665)). But the default parser doesn't include an `isDevVersion` method, so we have to add one by adding it to the manually-created super ([here](https://github.com/elastic/kibana/pull/192027/files#diff-b0a694370f3ac8a1408cc294df0c9277f3320dccb3f833b3a0a207723967c36fR12)).

Reply:
Ahhh TIL!